### PR TITLE
INT8 Pool kernel Key Creation Optimization.

### PR DESCRIPTION
**Pull request by xiaolil1**
Optimize key creation of INT8 pool kernel to improve the peformance of ResNet-50 and MobileNet, especially for latency.

test=develop

### CODE REVIEW

**Comment by xiaolil1** [comment]
@kbinias @sfraczek @wojtuss  Please help to review this PR, thanks!

File: paddle/fluid/operators/mkldnn/pool_mkldnn_op.cc [hunk 1]
```diff
@@ -29,23 +30,23 @@ using mkldnn::stream;
...
-  auto dims2str = [](const memory::dims& operand_dims) {
-    std::string dstr = "";
-    for (size_t i = 0; i < operand_dims.size(); ++i) {
-      dstr += std::to_string(operand_dims[i]) + "-";
-    }
-    return dstr;
-  };
-  return dims2str(input_dims) + dims2str(ksize) + dims2str(strides) +
-         dims2str(paddings) + std::to_string(dt) + pooling_type + suffix;
+std::string CreateKey(const paddle::framework::ExecutionContext& ctx,
```
Comment (sfraczek) [inline paddle/fluid/operators/mkldnn/pool_mkldnn_op.cc:33]: I wonder if there was any particular reason why you have removed the static keyword here?

**Review by sfraczek — APPROVED** [review]
LGTM. I just have one comment (see above).

File: paddle/fluid/operators/mkldnn/pool_mkldnn_op.cc (hunk 1 above)
Comment (hshen14) [reply paddle/fluid/operators/mkldnn/pool_mkldnn_op.cc:33]: Thanks @sfraczek. We thought the current design on hash key function depends on OP implementation with MKL-DNN kernel and such key is only visible to the corresponding op, so we removed the static qualifier to make it simpler.

**Pull request merged**